hazelcast/src/test/java/com/hazelcast/client/map/impl/nearcache/ClientMapNearCacheInvalidationTest.java

```diff
@@ -158,6 +169,155 @@ public void testMapClear_shouldClearNearCaches_onOwnerAndBackupNodes() {
         assertNearCacheSizeEventually(clientMap, 0);
     }
 
+
+
+    @Test
+    public void testMapSubmitToKey_withReadOnlyProcessor_noInvalidations() {
+        Config config = getConfig();
+        configureBatching(config, 10, 1);
+
+        ClientConfig clientConfig = getClientConfig(mapName);
+
+        HazelcastInstance server = factory.newHazelcastInstance(config);
+        factory.newHazelcastInstance(config);
+
+        HazelcastInstance client = factory.newHazelcastClient(clientConfig);
+        makeSureConnectedToServers(client, 1);
+
+        IMap<Integer, Integer> serverMap = server.getMap(mapName);
+        IMap<Integer, Integer> clientMap = client.getMap(mapName);
+
+        int size = 1000;
+
+        // fill serverMap
+        for (int i = 0; i < size; i++) {
+            serverMap.put(i, i);
+        }
+
+        NearCacheStats nearCachestats = ((NearCachedClientMapProxy) clientMap).getNearCache().getNearCacheStats();
+        long invalidationsBefore = nearCachestats.getInvalidations();
+        // fill Near Cache on client
+        for (int i = 0; i < size; i++) {
+            clientMap.submitToKey(i, new TestReadOnlyProcessor());
+        }
+
+        long invalidationsAfter = nearCachestats.getInvalidations();
+        assertEquals("No Invalidation after getting keys from read only entry processor ", invalidationsBefore, invalidationsAfter);
+
+    }
+
+
+
+    @Test
+    public void testMapExecuteToKey_withReadOnlyProcessor_noInvalidations() {
```

Comment:
The same comments above also apply to this test. Also, the new tests in this file are almost the same. Can you try to get rid of the duplicate codes? You could have a method named `validateNoInvalidationsWithReadOnlyEntryProcessor` and pass the different methods of sending a read-only entry processor.

Reply:
I have kept different tests in case we want to try some customisations in particular functions. This way it can be kept independent of other changes